src/components/EditAbilityScores.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import 'airbnb-js-shims' // to allow jest to understand Object.entries for parsing the character objects
+
+import { Form, FormGroup, Col, Button, ButtonGroup, ControlLabel } from 'react-bootstrap'
+
+export default class EditAbilityScores extends Component {
+  constructor (props) {
+    super(props)
+
+    this.renderAbilityScorePackages = this.renderAbilityScorePackages.bind(this)
+    this.pointsRemaining = this.pointsRemaining.bind(this)
+    this.increaseScore = this.increaseScore.bind(this)
+    this.decreaseScore = this.decreaseScore.bind(this)
+    this.resetScores = this.resetScores.bind(this)
+  }
+
+  pointsRemaining () {
+    let points = 27
+
+    Object.values(this.props.abilityScores).map((score) => {
```

Comment:
heh, here you're using a `map()` when really you want a `forEach` because you're not returning anything from the body of the `map` callback, and you're not assigning the value of the `map` statement to anything. In this case, it would be preferable to just use a `forEach()`

*however* this is actually a perfect case for the use of `reduce()` instead of `map()` or `forEach()`:

```
const STARTING_POINTS = 27
...
return Object.values(ABILITY_SCORES).reduce((pointsLeft, score) => {
  let delta = (score - 8)
  if (score === 15) {
    delta = 9
  } else if (score === 14) {
    delta = 7
  }
  return pointsLeft - delta
}, STARTING_POINTS)
```